presentation/virtualKeyboard.brs

```diff
@@ -3,41 +3,92 @@ REM @title                   BrightSign Virtual Keyboard
+        else
+          vidmode = createObject("roVideoMode")
+        end if
+        
+        width = vidmode.getResX()
+        height = vidmode.getResY()
+
+        ' rectSizeX and rectSizeY are the dimensions of the keyboard
+        rectSizeX = Cint(width*.3125) ' Using percentage adjustments based on the resolution allows for correct sizing regardless of res
+        rectSizeY = Cint(height*.37)
+        ' startCoordX and startCoordY are the coordinates of the top left corner of the keyboard
+        startCoordX = Cint((width-rectSizeX)/2) ' Placing the keyboard on the bottom of the screen. Using the dimensions of the keyboard guarantee that the entire rectangle will be rendered onscreen
+        startCoordY = Cint(height-rectSizeY)
+
+        ' Must fetch the monitor orientation to determine how to place and rotate the keyboard
+        monitorOrientation = ""
+        if m.bsp.sign.screens <> invalid then ' account for multiple screens
+          colCount = m.bsp.sign.screens.count()
+          if m.bsp.sign.screens[0] <> invalid then
+            rowCount = m.bsp.sign.screens[0].count()
+            if colCount > 1 or rowCount > 1 then
+              print "virtualKeyboard_Initialize: warning - confirm the keyboard orientation is correct per screen"
+              monitorOrientation = m.bsp.sign.screens[0].monitorOrientation
+              ' Here is where you would handle multiscreen keyboard orientation instances as needed
+            else if rowCount > 1 and rowCount > 1 then
+              monitorOrientation = m.bsp.sign.screens[0][0].monitorOrientation 
+            else
+              monitorOrientation = m.bsp.sign.monitorOrientation
+            end if
+          else
+            monitorOrientation = m.bsp.sign.monitorOrientation
+          end if
+        else
+          monitorOrientation = "landscape"
+        end if
+
+        ' Place, size and rotate the keyboard based on the monitor/tv orientation
+        if monitorOrientation = "portrait" then
+          startCoordXPortrait = Cint(width-rectSizeY)
+          startCoordYPortrait = Cint((height-rectSizeX)/2)
+          rectangle = createobject("rorectangle",startCoordXPortrait,startCoordYPortrait,rectSizeX,rectSizeY)
+          transform = "rot90"
+        else if monitorOrientation = "portraitbottomleft" then
+          rectangle = createobject("rorectangle",rectSizeX,rectSizeY/2,rectSizeX,rectSizeY)
+          transform = "rot270"
+        else if monitorOrientation = "portraitbottomonright" then
+          rectangle = createobject("rorectangle",rectSizeX,rectSizeY/2,rectSizeX,rectSizeY)
+          transform = "rot90"
+        else if monitorOrientation = "portraitbottomright" then
+          rectangle = createobject("rorectangle",rectSizeX,rectSizeY/2,rectSizeX,rectSizeY)
+          transform = "rot90"
+        else ' landscape
+          rectangle = createobject("rorectangle",startCoordX,startCoordY,rectSizeX,rectSizeY)
+          transform = ""
+        endif
+
```

Comment:
Now that I know brightsign uses the Left Hand Rule and not RHR, these rotation values seem incorrect to me. I believe I had tested this, but similarly to the above comment it may not have been thorough enough, and will re-run the tests with the new info in mind

Reply:
The rotation values were flipped for bottom left and bottom right, that has been fixed